src/snowflake/snowpark/session.py

```diff
@@ -722,12 +722,24 @@ def _resolve_packages(
             package_name = package_req.key
             if validate_package:
                 if package_name not in valid_packages:
+                    is_anaconda_terms_acknowledged = self._run_query(
+                        "select system$are_anaconda_terms_acknowledged()"
+                    )[0][0]
+                    if is_anaconda_terms_acknowledged:
```

Comment:
Actually it's not easy to test because in the test I'm not able to alter parameters :(. Please comment if you can come up with a way to test this error message.